### PR TITLE
Add optimization to norm for common norms

### LINKED ISSUE #5671: torch.norm(x-y, 2, 1) is significantly slower than torch.sqrt((x - y).pow(2).sum(1))
I've tested a few other norms as well:
```
x = torch.randn(1024, 256)
y = torch.randn(1024, 256)


In [12]: %timeit torch.norm(x-y, 1, 1)
2.55 ms ± 253 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [13]: %timeit (x-y).sum(1)
339 µs ± 699 ns per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [14]: %timeit torch.norm(x-y, 2, 1)
2.42 ms ± 33.7 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [15]: %timeit torch.sqrt((x-y).pow(2).sum(1))
736 µs ± 2.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [16]: %timeit torch.pow((x - y).pow(3).sum(1), 1/3)
700 µs ± 571 ns per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [17]: %timeit torch.norm(x-y, 3, 1)
16.1 ms ± 31.5 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

```
Reported [here on the forums](https://discuss.pytorch.org/t/torch-norm-3-6x-slower-than-manually-calculating-sum-of-squares/14684)

### PR DESCRIPTION
**Pull request by zou3519**
Fixes #5671 

The reported issue is that on the CPU path, norm(value, dim) is slower than manually using pow, sqrt, and summing.

It turns out that the CPU path for norm(value, dim) is missing optimizations in the `value=1, 2` cases. I added those in as well as an optimization for `value = 3` (not sure if this is necessary, but this optimization is used for `tensor.pow(3)`).

@li-roy could you take a look?

Perf numbers:
```

In [1]: import torch
   ...: x = torch.randn(1024, 256)
   ...: y = torch.randn(1024, 256)
   ...:
   ...: %timeit torch.norm(x-y, 1, 1)
   ...: %timeit (x-y).sum(1)
   ...:
   ...: %timeit torch.norm(x-y, 2, 1)
   ...: %timeit torch.sqrt((x-y).pow(2).sum(1))
   ...:
   ...: %timeit torch.norm(x-y, 3, 1)
   ...: %timeit torch.pow((x - y).abs().pow(3).sum(1), 1/3)
   ...:
362 µs ± 56.4 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
332 µs ± 33.2 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

340 µs ± 8.42 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
358 µs ± 5.87 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

352 µs ± 4.55 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
691 µs ± 49.2 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

### CODE REVIEW

**Review by apaszke — APPROVED** [review]
Does CUDA have optimized implementations?

**Comment by zou3519** [comment]
Yup!

**Comment by JaeDukSeo** [comment]
wait so is this fixed...?

**Comment by soumith** [comment]
@JaeDukSeo yes
